pkg/datapath/loader/base.go

```diff
@@ -343,10 +343,10 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 	}
 	args[initArgMode] = string(mode)
 
-	if option.Config.Tunnel == option.TunnelDisabled && option.Config.EnableIPv4EgressGateway {
+	if !option.Config.TunnelingEnabled() && option.Config.EnableIPv4EgressGateway {
 		// Enable tunnel mode to vxlan if egress gateway is configured
 		// Tunnel is required for egress traffic under this config
-		args[initArgTunnelMode] = option.TunnelVXLAN
+		args[initArgTunnelMode] = option.Config.TunnelProtocol
```

Comment:
The comment is no longer accurate (the protocol can now be either `vxlan` or `geneve`).